packages/typescript-estree/tests/snapshots/typescript/errorRecovery/interface-property-with-default-value.src.ts.shot

```diff
@@ -8,26 +8,6 @@ Object {
         "body": Array [
           Object {
             "computed": false,
-            "export": undefined,
-            "initializer": Object {
```

Comment:
hmm - i wonder if we should throw on this specific case?
[prettier currently supports it](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLWcBOAzAhmOAAgDEIJDgAdKQwgDyUIGcZN0BzQgXlt78IDkAgNzUAvtRAAaEBAAOMVNCbJQuTJggB3AArqEKlLgA2W3AE8VMgEaZ8AazgwAyrgC2cADLo4yPMaY4GzswRxc5fA5kVgBXIJA4N2s4ABMU1M9cKHYY3HY4Ukw3XBhFbOQQXBiYCGkQAAsYN2MAdXrUeCYIgmcDDtQANw7zCrAmKzQoQMwYHTt2Yr8TQJkAKyY6Zw5jOABFGIh4JYD4iMxpiphzOTgmMDYFOrk2WBbUFJh65AAOAAYZZ4QQItOxyCrPW5YAa+GQARwO8Dm8kMlSYAFooHBUqk6pg4PDUHi5nlFkh-CsQIE3Khopg4jImNs4ABBUpsazVOA6LDeTHHCmM7I7faHXxk5bxGC4axvD5fJAAJhkrFwqGMHAAwhA3KSEkwAKx1GKBAAq0sM5PiAziAEkoOlYM57qgFMz7c4rjt+XAxGIgA), but other parsers (babel) throw here.

If we don't throw, I believe prettier will just delete the code silently when it formats.
WDYT?

Reply:
https://github.com/prettier/prettier/pull/14264

Reply:
@bradzacher @fisker I went ahead and added errors for the other `@deprecated` node properties in `typescript.d.ts` (TIL!):

```ts
interface PropertySignature {
    /** @deprecated A property signature cannot have an initializer */
    readonly initializer?: Expression | undefined;
}
interface PropertyAssignment {
    /** @deprecated A property assignment cannot have a question token */
    readonly questionToken?: QuestionToken | undefined;
    /** @deprecated A property assignment cannot have an exclamation token */
    readonly exclamationToken?: ExclamationToken | undefined;
}
interface ShorthandPropertyAssignment {
    /** @deprecated A shorthand property assignment cannot have modifiers */
    readonly modifiers?: NodeArray<Modifier> | undefined;
    /** @deprecated A shorthand property assignment cannot have a question token */
    readonly questionToken?: QuestionToken | undefined;
    /** @deprecated A shorthand property assignment cannot have an exclamation token */
    readonly exclamationToken?: ExclamationToken | undefined;
}
interface FunctionTypeNode {
    /** @deprecated A function type cannot have modifiers */
    readonly modifiers?: NodeArray<Modifier> | undefined;
}
```

https://github.com/typescript-eslint/typescript-eslint/pull/6243#issuecomment-1416905513 / d3c054fc0

Do you think this is reasonable for Prettier too? I'd be interested in sending a PR over if so, I haven't actually contributed to Prettier yet!

Reply:
Great! Saving us from checking those.

Reply:
@fisker I poked around in Prettier-land and couldn't figure out if there was anything else for me to change there. I'll defer to you. If you end up filing an issue in Prettier I'd love to tackle it. But it's a bit much for me on my own. 😅

Reply:
The invalid modifier check is here in this file   https://github.com/prettier/prettier/blob/next/src/language-js/parse/postprocess/typescript.js

Reply:
And tests are here https://github.com/prettier/prettier/blob/next/tests/format/misc/errors/typescript/modifiers/jsfmt.spec.js